rllib/algorithms/dqn/dqn_rainbow_rl_module.py

```diff
@@ -0,0 +1,185 @@
+        # during exploration.
+        self.action_dist_cls = catalog.get_action_dist_cls(framework=self.framework)
+
+    @override(RLModule)
+    def get_exploration_action_dist_cls(self) -> Type[Distribution]:
+        """Returns the action distribution class for exploration.
+
+        Note, this class is used to sample the exploit action during
+        exploration.
+        """
+        return self.action_dist_cls
+
+    # TODO (simon): DQN Rainbow does not support RNNs, yet.
+    @override(RLModule)
+    def get_initial_state(self) -> Any:
+        return {}
+
+    @override(RLModule)
+    def input_specs_exploration(self) -> SpecType:
+        return [SampleBatch.OBS]
+
+    @override(RLModule)
+    def input_specs_inference(self) -> SpecType:
+        return [SampleBatch.OBS]
+
+    @override(RLModule)
+    def input_specs_train(self) -> SpecType:
+        return [
+            SampleBatch.OBS,
+            SampleBatch.ACTIONS,
+            SampleBatch.NEXT_OBS,
+        ]
+
+    @override(RLModule)
+    def output_specs_exploration(self) -> SpecType:
+        return
+
+    @abstractmethod
+    @OverrideToImplementCustomLogic
+    def _qf(self, batch: Dict[str, TensorType]) -> Dict[str, TensorType]:
+        """Computes Q-values.
+
+        Note, these can be accompanied with logits and pobabilities
+        in case of distributional Q-learning, i.e. `self.num_atoms > 1`.
+
+        Args:
+            batch: The batch recevied in the forward pass.
+
+        Results:
+            A dictionary containing the Q-value predictions and in case of
+            distributional Q-leanring the support atoms ("atoms"), the
+            Q-logits ("qf_logits"), and the probabilities ("qf_probs").
+        """
+
+    @abstractmethod
+    @OverrideToImplementCustomLogic
+    def _qf_target(self, batch: Dict[str, TensorType]) -> Dict[str, TensorType]:
+        """Computes Q-values from the target network.
+
+        Note, these can be accompanied with logits and pobabilities
+        in case of distributional Q-learning, i.e. `self.num_atoms > 1`.
+
+        Args:
+            batch: The batch recevied in the forward pass.
+
+        Results:
+            A dictionary containing the target Q-value predictions and in
+            case of distributional Q-leanring the support atoms ("atoms"),
+            the  Q-logits ("qf_logits"), and the probabilities ("qf_probs").
+        """
+
+    @abstractmethod
+    @OverrideToImplementCustomLogic
+    def _af_dist(self, batch: Dict[str, TensorType]) -> Dict[str, TensorType]:
+        """Compute the advantage distribution.
+
+        Note this distribution is identical to the Q-distribution in
+        case no dueling architecture is used.
+
+        Args:
+            batch: A dictionary containing a tensor with the outputs of the
+                forward pass of the Q-head or advantage stream head.
+
+        Returns:
+            A `dict` containing the support of the discrete distribution for
```

Comment:
Same here. Let's describe all str keys required in the different case, such that there remains on ambiguity for the users that want to implement this themselves.